components/script/script_thread.rs

```diff
@@ -221,7 +222,7 @@ pub struct CancellableRunnable<T: Runnable + Send> {
 }
 
 impl<T: Runnable + Send> Runnable for CancellableRunnable<T> {
-    fn name(&self) -> &'static str { self.inner.name() }
+    fn name(&self) -> &'static str { unsafe { intrinsics::type_name::<Self>() } }
```

Comment:
I believe this should be in the definition of `trait Runnable`, and after that all the manual implementations should go away.

Reply:
makes sense. will implement the change